core/src/com/unciv/logic/battle/CityCombatant.kt

```diff
@@ -28,7 +28,7 @@ class CityCombatant(val city: CityInfo) : ICombatant {
         if (city.health < 1) city.health = 1  // min health is 1
     }
 
-    override fun getUnitType(): UnitType = UnitType.City
+    override fun getUnitType(): UnitType = UnitType("City")
```

Comment:
You could make this drop-in compatible by delivering ".City" from the companion object - might also be potentially more efficient as the instance will be reused - and I see no reason why it shouldn't. The new UnitType is no longer immutable due to deserialization, but functionally they're stable or the old code wouldn't have worked... Or am I thinking wrong?